docs/Spring Boot/5_Step_To_Make_Gradle_Configuration_Extreme_Clean_In_A_Multi-Module_Project.md

```diff
@@ -0,0 +1,216 @@
+# 5 Step to Make Gradle Configuration Extreme Clean in a Multi-Module Project
+
+Multi-module Gradle projects involve numerous tasks during the build process. Managing dependency version control, plugin usage, build logic, and more with Gradle proves to be a popular and effective approach. But, achieving these tasks requires a lot of configuration scripts, which can make the file more complicated, and more difficult for development. These steps in the article will guide you through a clean and efficient way to manage configuration files:
+
+1. extract version declaring in `gradle.properties`.
+2. define all plugin and repository in `settings.gradle`.
+3. define all library in the `allprojects.dependencyManagement` in `./build.gradle`.
+4. declaring dependency and plugin directly instead of using `subproject` in submodule.
+5. extract complex and common task config to extra file and apply wherever need it.
+
+Take a look at this [repository](https://github.com/NoahHsu/event-sourcing-order-poc) or [refactor PR](https://github.com/NoahHsu/event-sourcing-order-poc/pull/72), if you can't wait to find out how it looks.
+
+## Step by Step Demonstration
+
+### Step 1: Extract Version Declaration
+
+Version declarations can be extracted into a `gradle.properties` file. Additionally, Gradle arguments can be defined as shown below:
+
+```properties title="./gradle.properties"
+group='org.example'
+version=0.0.1.SNAPSHOT
+
+# Plugin Version
+jibVersion=3.4.3
+
+# Spring Version
+springBootVersion=3.1.5
+springDependencyVersion=1.1.4
+springCloudVersion=2022.0.1
+
+# Dependency Version
+springdocVersion=2.1.0
+feignMicrometerVersion=12.1
+wiremockVersion=3.7.0
+logbackAppenderVersion=1.4.0-rc2
+lombokVersion=1.18.20
+
+# Gradle Argument
+org.gradle.parallel=true
+```
+
+### Step 2: Define Used Plugins and Maven Source 
+
+All used plugins and the source Maven repository can be defined in a `settings.gradle`.:
```

Comment:
⚠️ **[vale]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>[write-good.Passive] 'be defined' may be passive voice. Use active voice if you can.